docs/docs/reference/changed-features/pattern-matching.md

```diff
@@ -62,12 +125,53 @@ object FirstChars {
 // First: H; Second: i
 ```
 
+## Single Match
+
+- If there is exactly `1` pattern, pattern-matching on `1` pattern with type `U`
+
+<!-- To be kept in sync with tests/new/patmat-spec.scala -->
+
+```scala
+class Nat(val x: Int) {
+  def get: Int = x
+  def isEmpty = x < 0
+}
+
+object Nat {
+  def unapply(x: Int): Nat = new Nat(x)
+}
+
+5 match {
+  case Nat(n) => println(s"$n is a natural number")
+  case _      => ()
+}
+// 5 is a natural number
+```
+
+## Name-based Match
+
+- `N > 1` is the maximum number of consecutive (parameterless `def` or `val`) `_1: P1 ... _N: PN` members in `U`
+- Pattern-matching on exactly `N` patterns with types `P1, P2, ..., PN`
+
+```Scala
+object ProdEmpty {
+  def _1: Int = ???
+  def _2: String = ???
+  def isEmpty = true
+  def unapply(s: String): this.type = this
+  def get = this
+}
+
+"" match {
+  case ProdEmpty(_, _) => ???
+  case _ => ()
+}
+```
+
 
-## Name-based Seq Pattern
+## Seq Match
```

Comment:
It's entirely satisfying that everything is name based expect for "Seq Match", were we still require the return type of `drop` and `toSeq` to conform to the actual `scala.Seq` instead of something which has the same shape. I think this is also a divergence from scalac as seen in the diff needed to move `run/string-extractor.scala` from `pending` to `run`...

Reply:
You mean we drop the `U <: Product` requirement. We can try that, but I suggest we do it in another PR.

The pending tests are ported from Scalac, there is no divergence:

https://github.com/scala/scala/commit/308ae2d3967cf885f2aa5c5c2489b79abb865d42#diff-f761b9d2bf9ec2f17bc00f552fc07314